tests/performance/src/test/java/org/mule/test/core/flow/FlowPerformanceTestCase.java

```diff
@@ -0,0 +1,186 @@
+    return asList(new Object[][] {
+        {new SynchronousProcessingStrategyFactory(), singletonList(liteProcessor)},
+        {new ReactorProcessingStrategyFactory(), singletonList(liteProcessor)},
+        {new MultiReactorProcessingStrategyFactory(), singletonList(liteProcessor)},
+        {new ProactorProcessingStrategyFactory(), singletonList(liteProcessor)},
+        {new RingBufferProcessingStrategyFactory(), singletonList(liteProcessor)},
+        {new ParellelProcessingStrategyFactory(), singletonList(liteProcessor)}
+    });
+  }
+
+  protected void addBuilders(List<ConfigurationBuilder> builders) {
+    builders.add(new BasicRuntimeServicesConfigurationBuilder());
+  }
+
+  @Before
+  public void setup() throws MuleException {
+    muleContext.start();
+
+    source = new TriggerableMessageSource();
+    flow = new Flow("flow", muleContext);
+    flow.setMessageProcessors(processors);
+    flow.setMessageSource(source);
+    flow.setProcessingStrategyFactory(processingStrategyFactory);
+    muleContext.getRegistry().registerFlowConstruct(flow);
+
+    fluxProcessor = EmitterProcessor.create(1);
+    fluxProcessor.transform(source)
+        .doOnError(EventDroppedException.class, mde -> mde.getEvent().getContext().success())
+        .doOnNext(response -> response.getContext().success(response))
+        .doOnError(MessagingException.class, me -> me.getEvent().getContext().error(me))
+        .subscribe();
+    fluxSink = fluxProcessor.connectSink();
+  }
+
+  @After
+  public void cleanup() throws MuleException {
+    fluxSink.complete();
+    ((DefaultSchedulerService) muleContext.getSchedulerService()).stop();
+  }
+
+  @Test
+  @PerfTest(duration = 15000, threads = 1, warmUp = 5000)
+  public void blocking() throws Exception {
+    for (int i = 0; i < ITERATIONS; i++) {
+      source.trigger(Event.builder(DefaultEventContext.create(flow, TEST_CONNECTOR))
+          .message(InternalMessage.of(TEST_PAYLOAD)).build());
+    }
+  }
+
+  @Test
+  @PerfTest(duration = 15000, threads = 1, warmUp = 5000)
+  public void mono() throws Exception {
+    for (int i = 0; i < ITERATIONS; i++) {
+      Mono.just(Event.builder(DefaultEventContext.create(flow, TEST_CONNECTOR)).message(InternalMessage.of(TEST_PAYLOAD))
+          .build()).transform(source).block(Duration.of(LOCK_TIMEOUT, MILLIS));
+    }
+  }
+
+  @Test
+  @PerfTest(duration = 15000, threads = 1, warmUp = 5000)
+  public void fluxPush() throws Exception {
+    CountDownLatch latch = new CountDownLatch(ITERATIONS);
+    for (int i = 0; i < ITERATIONS; i++) {
+      Event event = Event.builder(DefaultEventContext.create(flow, TEST_CONNECTOR))
+          .message(InternalMessage.of(TEST_PAYLOAD)).build();
+      fluxSink.accept(event);
+      Mono.from(event.getContext()).doOnNext(e -> latch.countDown()).subscribe();
+    }
+    latch.await(LOCK_TIMEOUT, SECONDS);
+  }
+
+  @Test
+  @PerfTest(duration = 15000, threads = 1, warmUp = 5000)
+  public void fluxPull() throws Exception {
```

Comment:
add jdoc. It's not at all clear what is being tested here. This also seems as something we should be adding Allure annotations to.